packages/beelzebub/data_stream/logs/elasticsearch/ingest_pipeline/default.yml

```
@@ -170,53 +170,73 @@ processors:
       extract_device_type: true
       ignore_missing: true
 
-  - gsub:
-      if: ctx.json?.event?.Headers != null && ctx.json?.event?.Headers != ""
-      description: "Replace leading and trailing garbage to support KVP extraction"
+### NOTE: Beelzebub now produces logs with two fields for headers, one is a
+#   string and the other is an object. The following 7 processors are used to
+#   support both log formats/situations and to avoid creating a type conflict
+#   with existing indices while still ensuring we have usable headers and
+#   header values to search with.
+
+  - remove:
+      if: ctx.json?.event?.Headers == ""
+      description: "Remove json.event.Headers if it is null or an empty string so that we can skip some of processors"
```

OK, so this needs to be adjusted then since the comment does not match the behaviour; this will not remove the field if it is explicitly `null`.
```suggestion
      description: "Remove json.event.Headers if it an empty string so that we can skip some of processors"
```